app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -448,23 +448,26 @@ object AppRepository {
         }
         val dayRanges = loadSessionsForAllDays(includeEngelsystemShifts = false)
                 .toDayRanges()
+        val oldShifts = loadEngelsystemShiftsForAllDays()
         val sessionizedShifts = shifts
                 .also { logging.d(LOG_TAG, "Shifts unfiltered = ${it.size}") }
                 .cropToDayRangesExtent(dayRanges)
                 .also { logging.d(LOG_TAG, "Shifts filtered = ${it.size}") }
                 .toSessionAppModels(logging, ENGELSYSTEM_ROOM_NAME, dayRanges)
                 .sanitize()
-        val toBeUpdatedSessions = loadSessionsForAllDays(false) // Drop all shifts before ...
+        val shiftChanges = computeSessionsWithChangeFlags(sessionizedShifts, oldShifts)
+        if (oldShifts.isEmpty() || shiftChanges.foundAnyChanges) {
+            val toBeUpdatedSessions = loadSessionsForAllDays(false) // Drop all shifts before ...
                 .toMutableList()
                 // Shift rooms to make space for the Engelshifts room
                 .shiftRoomIndicesOfMainSchedule(sessionizedShifts.toDayIndices())
```

Comment:
Unrelated to this pull request, but I'm curious. Won't this shift room indices by one each time the angel shifts are updated (first creating, then widening a gap)?

Reply:
:heavy_check_mark: Nice catch. Yes, that is the case currently - I checked it. Fixed now.